core/templates/pages/contributor-dashboard-page/login-required-message/login-required-message.component.ts

```diff
@@ -13,45 +13,56 @@
 // limitations under the License.
 
 /**
- * @fileoverview Component for the item view of an opportunity.
+ * @fileoverview Component for login required message.
  */
 
-require('services/site-analytics.service.ts');
-require('services/user.service.ts');
-
-angular.module('oppia').component('loginRequiredMessage', {
-  template: require('./login-required-message.component.html'),
-  controller: [
-    '$rootScope', '$timeout', '$window', 'SiteAnalyticsService',
-    'UrlInterpolationService', 'UserService',
-    'OPPIA_AVATAR_LINK_URL', function(
-        $rootScope, $timeout, $window, SiteAnalyticsService,
-        UrlInterpolationService, UserService,
-        OPPIA_AVATAR_LINK_URL) {
-      var ctrl = this;
-      ctrl.onLoginButtonClicked = function() {
-        UserService.getLoginUrlAsync().then(
-          function(loginUrl) {
-            if (loginUrl) {
-              SiteAnalyticsService.registerStartLoginEvent('loginButton');
-              $timeout(function() {
-                $window.location = loginUrl;
-              }, 150);
-            } else {
-              $window.location.reload();
-            }
-            // TODO(#8521): Remove the use of $rootScope.$apply()
-            // once the controller is migrated to angular.
-            $rootScope.$applyAsync();
-          }
-        );
-      };
-      ctrl.$onInit = function() {
-        ctrl.OPPIA_AVATAR_LINK_URL = OPPIA_AVATAR_LINK_URL;
-        ctrl.OPPIA_AVATAR_IMAGE_URL = (
-          UrlInterpolationService.getStaticImageUrl(
-            '/avatar/oppia_avatar_100px.svg'));
-      };
-    }
-  ]
-});
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { UserService } from 'services/user.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import constants from 'assets/constants';
+import { Component } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+@Component({
+  selector: 'login-required-message',
+  templateUrl: './login-required-message.component.html',
+  styleUrls: []
+})
+export class LoginRequiredMessageComponent {
+  OPPIA_AVATAR_IMAGE_URL: string;
+  OPPIA_AVATAR_LINK_URL: string;
+
+  constructor(
+    private siteAnalyticsService: SiteAnalyticsService,
+    private urlInterpolationService: UrlInterpolationService,
+    private userService: UserService,
+    private windowRef: WindowRef) {}
```

Comment:
Can be marked as `readonly`?

Reply:
It makes sense to me to mark them as readonly, but it doesn't look like the `private readonly` convention is used anywhere else in the codebase.
I think it's more consistent to stick with just `private`.

Reply:
Its use in the codebase might not be widespread, but I do see a few instances of its usage, e.g. in core/templates/domain/improvements/exploration-improvements-config.model.ts.

My 2 cents is that it would be good to use it where we can so we get the benefit of TypeScript compiler checks (in this case if someone tries to reassign a variable marked as `readonly`, [TypeScript would complain](https://www.typescriptlang.org/docs/handbook/classes.html#readonly-modifier)).

Reply:
I added the `readonly` modifier.
@srijanreddy98, adding `readonly` to injected services seems valuable in terms of typescript checks. Could this be included in the the Angular Migration wiki?